design/outlier-detection-design.md

```diff
@@ -0,0 +1,275 @@
+              splitExternalLocalOriginErrors: true
+              consecutiveLocalOriginFailure: 5
+```
+In this example, when the s1 service has 5 consecutive local origin errors, the s1 service will be ejected, and the 5 xx errors returned by the upstream service will not be counted in the error statistics. The maximum eviction ratio is 100%. Panic is disabled, which is also the default value.
+
+Below is the generated envoy configuration:
+```yaml
+cluster:
+  common_lb_config:
+    healthy_panic_threshold:
+      value: 0.0
+  outlier_detection:
+    enforcing_success_rate: 0
+    enforcing_consecutive_5xx: 0
+    max_ejection_percent: 100
+    split_external_local_origin_errors: true
+    consecutive_local_origin_failure: 5
+    #Default to 100% if not specified
+    #enforcing_consecutive_local_origin_failure: 100
+    enforcing_local_origin_success_rate: 0
+    enforcing_consecutive_gateway_failure: 0
+```
+
+### Example 3
+ ```yaml
+apiVersion: projectcontour.io/v1alpha1
+kind: HTTPProxy
+metadata:
+  name: local-external-origin-errors-and-consecutive-5xx
+  namespace: projectcontour
+spec:
+    virtualhost:
+        fqdn: outlierDetection.projectcontour.io
+    routes:
+      - conditions:
+          - prefix: /
+        services:
+          - name: s1
+            port: 80
+            outlierDetection:
+              consecutiveServerErrors: 10
+              maxEjectionPercent: 100
+              splitExternalLocalOriginErrors: true
+              consecutiveLocalOriginFailure: 5
+```
+In this example, when the s1 service continuously encounters 5 local origin errors or 10 external errors, the s1 service will be ejected. The maximum eviction rate is 100%. Panic is disabled, which is also the default value.
+
+Below is the generated envoy configuration:
+```yaml
+cluster:
+  common_lb_config:
+    healthy_panic_threshold:
+      value: 0.0
+  outlier_detection:
+    enforcing_success_rate: 0
+    consecutive_5xx: 10
+    enforcing_consecutive_5xx: 100
+    max_ejection_percent: 100
+    split_external_local_origin_errors: true
+    consecutive_local_origin_failure: 5
+    #Default to 100% if not specified
+    #enforcing_consecutive_local_origin_failure: 100
+    enforcing_local_origin_success_rate: 0
+    enforcing_consecutive_gateway_failure: 0
+``` 
+
+Notes:
+- If consecutiveServerErrors is specified as 0, and splitExternalLocalOriginErrors is true, then local errors will be ignored.This is especially useful when the upstream service explicitly returns a 5xx for some requests and you want to ignore those responses from upstream service while determining the outlier detection status of a host.
+- When accessing the upstream host through an opaque TCP connection, connection timeouts, connection errors, and request failures are all considered 5xx errors, and therefore these events are included in the 5xx error statistics.
+- Please refer to the [official documentation of Envoy][1] for more instructions.
+
```

Comment:
we should also include some discussion/documentation on the different behavior between http router and tcp filter when this is in the mix